src/main/java/jenkins/plugins/office365connector/Office365ConnectorWebhookNotifier.java

```diff
@@ -242,16 +214,15 @@ private Card createJobCompletedCard() {
             Run previousBuild = run.getPreviousBuild();
             Result previousResult = (previousBuild != null) ? previousBuild.getResult() : Result.SUCCESS;
             AbstractBuild failingSinceRun = null;
-            Run rt;
-            rt = run.getPreviousNotFailedBuild();
+            Run rt = run.getPreviousNotFailedBuild();
             try {
                 if (rt != null) {
                     failingSinceRun = (AbstractBuild) rt.getNextBuild();
                 } else {
                     failingSinceRun = (AbstractBuild) run.getParent().getFirstBuild();
                 }
-            } catch (Throwable e) {
-                //listener.getLogger().println(e.getMessage());
+            } catch (ClassCastException e) {
```

Comment:
will investigate later if this can even happen but for now I catch the only exception that makes sense

Reply:
Am not sure if this is fine. If we are not able to get the failingSinceRun, we should not fail to send the card. We should eat whatever the exception and still send the card.

Reply:
This won't fail sending the card. We are simply just logging the exception and continuing.

Reply:
Perhaps instead of using the build log to print the error.
We should user a static class logger.

Reply:
Oh, I ran into on Pipeline build now that it's being logged 👍

Reply:
Can be rewritten to not try and cast it... and Actually just use [`Run`](http://javadoc.jenkins-ci.org/hudson/model/Run.html) which [`AbstractBuild`](http://javadoc.jenkins-ci.org/hudson/model/AbstractBuild.html) and [`WorkflowRun`](http://javadoc.jenkins.io/plugin/workflow-job/org/jenkinsci/plugins/workflow/job/WorkflowRun.html) is child to.

Reply:
```
            Run rt = run.getPreviousNotFailedBuild();
            Run failingSinceRun;
            if (rt != null) {
                failingSinceRun = rt.getNextBuild();
            } else {
                failingSinceRun = run.getParent().getFirstBuild();
            }
```

Reply:
Sorted it on #22 https://github.com/jenkinsci/office-365-connector-plugin/pull/22/files#diff-c8d3ccb3e6896886dafa3f9232f4f15cR207